tests/h/services/group_test.py

```diff
@@ -5,344 +5,408 @@
 
-    def test_create_private_group_sets_group_name(self, service):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_sets_group_name(self, creator, svc):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
         assert group.name == 'Anteater fans'
 
-    def test_create_private_group_sets_group_authority(self, service):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_sets_group_authority(self, svc, creator, pyramid_request):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
-        assert group.authority == 'example.com'
+        assert group.authority == pyramid_request.authority
 
-    def test_create_private_group_sets_group_creator(self, service, users):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_sets_group_creator(self, svc, creator):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
-        assert group.creator == users['cazimir']
+        assert group.creator == creator
 
-    def test_create_private_group_sets_description_when_present(self, service):
-        group = service.create_private_group('Anteater fans', 'cazimir', 'all about ant eaters')
+    def test_create_private_group_sets_description_when_present(self, svc, creator):
+        group = svc.create_private_group('Anteater fans', creator.userid, 'all about ant eaters')
 
         assert group.description == 'all about ant eaters'
 
-    def test_create_private_group_skips_setting_description_when_missing(self, service):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_skips_setting_description_when_missing(self, svc, creator):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
         assert group.description is None
 
-    def test_create_private_group_adds_group_creator_to_members(self, service, users):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_adds_group_creator_to_members(self, svc, creator):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
-        assert users['cazimir'] in group.members
+        assert creator in group.members
 
     @pytest.mark.parametrize('flag,expected_value', [
         ('joinable_by', JoinableBy.authority),
         ('readable_by', ReadableBy.members),
         ('writeable_by', WriteableBy.members)])
-    def test_create_private_group_sets_access_flags(self, service, flag, expected_value):
-        group = service.create_private_group('Anteater fans', 'cazimir')
+    def test_create_private_group_sets_access_flags(self, svc, creator, flag, expected_value):
+        group = svc.create_private_group('Anteater fans', creator.userid)
 
         assert getattr(group, flag) == expected_value
```

Comment:
Yet another comment on the pre-existing code but this seems like possibly a silly use of parametrize to me. Why not just put three asserts in the method instead?

Reply:
(There's a few parametrizes like this in the file.)